src/bthome_ble/parser.py

```diff
@@ -295,48 +295,77 @@ def _parse_payload(self, payload: bytes) -> bool:
         payload_length = len(payload)
         next_obj_start = 0
         result = False
+        measurements: list[dict[str, Any]] = []
+        device_id_dict: dict[int, int] = {}
 
+        # Create a list with all individual objects
         while payload_length >= next_obj_start + 1:
             obj_start = next_obj_start
-
             obj_control_byte = payload[obj_start]
-            obj_meas_type = payload[obj_start + 1]
             obj_data_length = (obj_control_byte >> 0) & 31  # 5 bits (0-4)
             obj_data_format = (obj_control_byte >> 5) & 7  # 3 bits (5-7)
+            obj_meas_type = payload[obj_start + 1]
 
-            obj_end = obj_start + obj_data_length
-            next_obj_start = obj_end + 1
+            if obj_data_length == 0:
+                continue
 
             if payload_length < next_obj_start:
                 _LOGGER.debug("Invalid payload data length, payload: %s", payload.hex())
                 break
 
-            if obj_data_length == 0:
-                continue
+            obj_data_start = obj_start + 2
+            obj_end = obj_start + obj_data_length
+            next_obj_start = obj_end + 1
+            measurements.append(
+                {
+                    "data format": obj_data_format,
+                    "data length": obj_data_length,
+                    "measurement type": obj_meas_type,
+                    "measurement data": payload[obj_data_start:next_obj_start],
+                    "device id": None,
+                }
+            )
 
-            if obj_meas_type not in MEAS_TYPES:
+        # Get a list of measurement types that are included more than once.
+        meas_list = []
+        for meas in measurements:
+            meas_list.append(meas["measurement type"])
+        dup_meas_types = {x for x in meas_list if meas_list.count(x) > 1}
```

Comment:
This is more efficient:
```suggestion
        seen_meas_types = set()
        dup_meas_types = set()

        for meas in measurements:
            if meas["measurement type"] in seen_meas_types:
                dup_meas_types.add(meas["measurement type"])
            else:
                seen_meas_types.add(meas["measurement type"])
```

Reply:
fixed according to suggestion